src/core/swap.c

```diff
@@ -1237,7 +1237,7 @@ static Unit *swap_following(Unit *u) {
         }
 
         /* Fall back to the first on the list */
-        return UNIT(first);
+        return first ? UNIT(first) : NULL;
```

Comment:
Let's fix `UNIT` definition instead and avoid the member access if the argument is NULL.